mainFunctionUpdates.py

```diff
@@ -14,94 +14,111 @@
-        #         html.Label("Select Timeframe", className="fw-bold me-2"),
-        #         dcc.Dropdown(
-        #             id='timeframe-dropdown',
-        #             options=[
-        #                 {'label': 'Previous 7 Days', 'value': 7},
-        #                 {'label': 'Previous 2 Weeks', 'value': 14},
-        #                 {'label': 'Previous 30 Days', 'value': 30},
-        #                 {'label': 'Previous 6 Months', 'value': 180},
-        #                 {'label': 'Previous Year', 'value': 365}
-        #             ],
-        #             value=7,
-        #             clearable=False,
-        #             style={'backgroundColor': '#fcfcfc', 'color': 'black'}
-        #         )
-        #     ], xs=12, sm=12, md=6, lg=4, xl=3, className="mb-4")
-        # ], justify="left"),
-        
-        # First row: Map
+
         dbc.Row([
             dbc.Col([
-                dcc.Graph(id='map-graph', figure=map_fig, responsive=True)
-            ], className="mb-4")
-        ]),
-           # Second row: Mobile sparklines
+                html.Label("Select Timeframe", className="fw-bold me-2"),
+                dcc.Dropdown(
+                    id='timeframe-dropdown',
+                    options=[
+                        {'label': 'Previous 7 Days', 'value': 7},
+                        {'label': 'Previous 2 Weeks', 'value': 14},
+                        {'label': 'Previous 30 Days', 'value': 30},
+                        {'label': 'Previous 6 Months', 'value': 180},
+                        {'label': 'Previous Year', 'value': 365}
+                    ],
+                    value=7,
+                    clearable=False,
+                    style={'backgroundColor': '#fcfcfc', 'color': 'black'}
+                )
+            ], xs=12, sm=12, md=6, lg=4, xl=3, className="mb-4")
+        ], justify="left"),
+        
+
         dbc.Row([
             dbc.Col([
-                dcc.Graph(id='mobile-sparklines', figure=mobile_sparklines_fig, responsive=True)
+                dcc.Graph(id='map', figure=map_fig, responsive=True)
             ], className="mb-4")
         ]),
-     # Third row: Bar Chart
         dbc.Row([
             dbc.Col([
-                dcc.Graph(id='bar-graph', figure=bar_fig, responsive=True)
-            ], width=12, className="mb-4")
+                dcc.Graph(id='line-chart', figure=line_fig, responsive=True)
+            ], width=6, className="mb-4"),
+            dbc.Col([
+                dcc.Graph(id='bar-chart', figure=bar_fig, responsive=True)
+            ], width=6, className="mb-4")
         ]),
         ]),
         ])
 
-# # Create callback 
-# @callback(
-#     Output(component_id='map-graph', component_property='figure'),
-#     Input(component_id='timeframe-dropdown', component_property='value'),
-#         )
+@app.callback(
+    Output('map', 'figure'),
+    Input('timeframe-dropdown','value')
+)
+def update_X_axis(number_of_days):
```

Comment:
@njsmithfm Rename this callback function please. It is not just updating the xaxis, but the rest of the chart.